packages/server/src/instances/monitoring.ts

```diff
@@ -10,24 +11,27 @@ const MAX_ALLOWED_FAILURES = 5
 export class InstanceMonitoring {
   constructor(
     private logger: Logger,
+    private distributed: DistributedService,
     private channels: ChannelService,
     private conduits: ConduitService,
     private instances: InstanceService,
     private failures: { [conduitId: string]: number }
   ) {}
 
-  async setup() {
-    await this.tickMonitoring()
-
-    setInterval(this.tickMonitoring.bind(this), ms('15s'))
+  async monitor() {
+    void this.tickMonitoring()
   }
 
   private async tickMonitoring() {
     try {
-      await this.initializeOutdatedConduits()
-      await this.loadNonLazyConduits()
+      await this.distributed.using('lock_instance_monitoring', async () => {
+        await this.initializeOutdatedConduits()
+        await this.loadNonLazyConduits()
+      })
     } catch (e) {
       this.logger.error('Error occurred while monitoring.', e.message)
+    } finally {
+      setTimeout(this.tickMonitoring.bind(this), ms('15s'))
```

Comment:
Aren't we exposed to problems if `initializeOutdatedConduits` and `loadNonLazyConduits` take more than 15sec to complete?

Reply:
No the locking will throw an exception and the node will just skip monitoring for that time. We just want to avoid multiple nodes doing the monitoring at the same time